services/src/main/java/org/keycloak/storage/openshift/OpenshiftClientStorageProvider.java

```diff
@@ -73,6 +75,12 @@ public ClientModel getClientByClientId(String clientId, RealmModel realm) {
         return new OpenshiftSAClientAdapter(clientId, resource, client, session, realm, providerModel);
     }
 
+    @Override
+    public List<ClientModel> searchClientsByClientId(String clientId, Integer firstResult, Integer maxResults, RealmModel realm) {
+        // TODO not sure about this, but I don't see this implementation using the search now
+        return Collections.singletonList(getClientByClientId(clientId, realm));
```

Comment:
@saibot94 @stianst I'm not sure about this. Does this mean, if we want to select a Client e.g. in some Client Role selector and this Client is an Openshift Client, we need to exactly match the Client ID to select it?

Reply:
I'm okay with leaving it not supporting search for now. The OpenshiftClientStorageProvider is not supported at the moment and most likely needs a bit of work regardless when we pick up the effort to be able to replace IdP in OpenShift with Keycloak.